application/src/main/java/org/togetherjava/tjbot/commands/free/ChannelMonitor.java

```diff
@@ -0,0 +1,245 @@
+     */
+    public boolean isChannelInactive(@NotNull final TextChannel channel) {
+        if (!channelsToMonitor.containsKey(channel.getIdLong())) {
+            throw new IllegalArgumentException(
+                    "Channel requested %s is not monitored by free channel"
+                        .formatted(channel.getName()));
+        }
+        // todo change the entire inactive test to work via restactions
+        return channel.getHistory()
+            .retrievePast(1)
+            .map(messages -> messages.get(0))
+            .map(Message::getTimeCreated)
+            .map(createdTime -> createdTime.isBefore(Util.anHourAgo()))
+            .complete();
+    }
+
+    /**
+     * This is a delegation method to pass operations to {@link ChannelStatus}. see
+     * {@link ChannelStatus#setBusy(long)} for details.
+     * 
+     * @param channelId the id for the channel status to modify.
+     * @param userId the id of the user changing the status to busy.
+     */
+    public void setChannelBusy(final long channelId, final long userId) {
+        channelsToMonitor.get(channelId).setBusy(userId);
+    }
+
+    /**
+     * This is a delegation method to pass operations to {@link ChannelStatus}. see
+     * {@link ChannelStatus#setFree()} for details.
+     * 
+     * @param channelId the id for the channel status to modify.
+     */
+    public void setChannelFree(final long channelId) {
+        channelsToMonitor.get(channelId).setFree();
+    }
+
+    /**
+     * This method provides a stream of the id's for guilds that are currently being monitored. This
+     * is streamed purely as a simple method of encapsulation.
+     *
+     * @return a stream of guild id's
+     */
+    public Stream<Long> guildIds() {
+        return postStatusInChannel.keySet().stream();
+    }
+
+    /**
+     * This method provides a stream of the id's for channels that status's are displayed in. This
+     * is streamed purely as a simple method of encapsulation.
+     * 
+     * @return a stream of channel id's
+     */
+    public Stream<Long> statusIds() {
+        return postStatusInChannel.values().stream();
+    }
+
+    private @NotNull List<ChannelStatus> guildMonitoredChannelsList(@NotNull final Guild guild) {
+        return guild.getChannels()
+            .stream()
+            .map(GuildChannel::getIdLong)
+            .filter(channelsToMonitor::containsKey)
+            .map(channelsToMonitor::get)
+            .toList();
+    }
+
+    /**
+     * Creates the status message (specific to the guild specified) that shows which channels are
+     * busy/free.
+     * <p>
+     * It gets the list of all channels in the guild and filters out all channels not being
+     * monitored (to get the correct channel order) it then updates the names of the channels in
+     * case they were changed on the guild.
+     * <p>
+     * If then iterates through all channels and checks which category they are in so that it can
```

Comment:
It, not If?

In general, try to not document implementation but behavior.

Reply:
Fixed ?